src/relations/db.py

```diff
@@ -414,23 +427,13 @@ def _on_relation_broken(self, broken_event: RelationBrokenEvent):
                     # we can guarantee this is the last one.
                     delete_db = False
                     break
-
         if delete_db:
             del cfg["databases"][database]
-            cfg["databases"].pop(f"{database}_standby")
+            cfg["databases"].pop(f"{database}_standby", None)
 
-        # delete user
-        self.charm.remove_user(user, cfg=cfg, render_cfg=True, reload_pgbouncer=True)
-        try:
-            if self.charm.backend_postgres:
-                # Try to delete user if backend database still exists. If not, postgres has been
-                # removed and will handle user deletion in its own relation-broken hook.
-                self.charm.backend_postgres.delete_user(user, if_exists=True)
-        except PostgreSQLDeleteUserError:
-            blockedmsg = f"failed to delete user for {self.relation_name}"
-            logger.error(blockedmsg)
-            self.charm.unit.status = BlockedStatus(blockedmsg)
-            return
+        cfg.remove_user(user)
+        self.charm.render_pgb_config(cfg, reload_pgbouncer=True)
+        self.charm.backend.postgres.delete_user(user)
```

Comment:
should there be some error handling here?

Reply:
We're already checking backend.postgres exists in line 416, so I'm not sure this is necessary. If we can't access it despite having the credentials, I'd argue that's an issue in postgres.